src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java

```diff
@@ -61,25 +67,32 @@ public CompanyProfileService(Logger logger, CompanyProfileRepository companyProf
      */
     public void updateInsolvencyLink(final CompanyProfile companyProfileRequest)
             throws NoSuchElementException {
-        Optional<CompanyProfileDocument> companyProfileOptional = companyProfileRepository
-                .findById(companyProfileRequest.getData().getCompanyNumber());
-
-        if (companyProfileOptional.isEmpty()) {
-            throw new NoSuchElementException("Database entry not found");
-        }
-
-        CompanyProfileDocument companyProfile = companyProfileOptional.get();
-        String insolvencyLink = companyProfileRequest.getData().getLinks().getInsolvency();
-        companyProfile.companyProfile.getLinks().setInsolvency(insolvencyLink);
-        companyProfileRepository.save(companyProfile);
-        logger.trace(String.format("DSND-376: Insolvency links updated: %s",
-                companyProfileRequest));
-
         String companyNumber = companyProfileRequest.getData().getCompanyNumber();
+        Query updateCriteria = new Query(Criteria.where("data.company_number").is(companyNumber));
+        Update updateQuery = new Update();
+        updateQuery.set("data.links.insolvency",
+                companyProfileRequest.getData().getLinks().getInsolvency());
+        updateQuery.set("data.etag",
+                GenerateEtagUtil.generateEtag());
+        UpdateResult updateResult = mongoTemplate.updateFirst(updateCriteria,
+                updateQuery,
+                "company_profile");
 
-        insolvencyApiService.invokeChsKafkaApi(companyNumber);
+        if (updateResult.getModifiedCount() == 1) {
+            logger.trace(String.format("DSND-376: Insolvency links updated for company number: %s",
+                    companyNumber));
+            try {
+                insolvencyApiService.invokeChsKafkaApi(companyNumber);
+                logger.info(String.format("DSND-377: ChsKafka api invoked successfully for company "
+                        + "number %s", companyNumber));
+            } catch (Exception exception) {
+                logger.info(String.format("Error invoking ChsKafka API for company number %s",
+                        companyNumber));
```

Comment:
guessing it's needed even if the invoke itself has a try catch inside?

Reply:
Yes, a log to trace if kafka message failed